### DISCUSSION TITLE
allow setting --branch or --alias flags when deploying

### DISCUSSION THREAD

**Discussion by zanona** [vercel/vercel#5228] [Ideas]
I am building a project straight from a GitHub action workflow, and, once the build passes, I am deploying to Vercel through the CLI. With that in mind, I was looking to pass the branch I am deploying the project or at least define an alias by the time of deployment, like so?

```sh
vercel --branch $GIT_BRANCH
#or
vercel --alias "$GIT_BRANCH--app.scope.vercel.app"
``` 

Currently, I am achieving the above by doing:

```sh
 npx vercel | { read url; npx vercel alias set "$url" "$GIT_BRANCH--app.scope.vercel.app"; }
```

The suggestion is really to make CLI users lives easier, and, adding the ability to define the branch related to the deployment, so, on Vercel UI, users can see to which branch that deploy is referring to.

**Comment by senecolas** [comment]
+1,
I dig up this idea from 2020 because I think it could be interesting.
With a self-hosted gitlab, we can't manage domains automatically with CI/CD Pipelines. This could be a good alternative of the domains-branch management dashboard.
This is also what is asked in the discussion of ticket #4853 

For the moment I have to use : 
```cmd
DEPLOYMENT_URL=$(vercel deploy --prebuilt --token=$VERCEL_TOKEN)
vercel alias $DEPLOYMENT_URL $PREVIEW_URL --token=$VERCEL_TOKEN --scope=$VERCEL_SCOPE
```
